src/main/java/io/github/christophermanahan/carnitas/Router.java

```diff
@@ -1,7 +1,9 @@
 package io.github.christophermanahan.carnitas;
```

Comment:
It feels like this class has multiple responsibilities. 
1) Matching routes
2) Constructing specific responses (ex: `methodNotAllowed` and `notFound`)

It may make sense to break these up